packages/app-desktop/gui/DefaultEditorDialog/Dialog.tsx

```diff
@@ -0,0 +1,231 @@
+	onClose: Function;
+}
+
+interface EditorInfo {
+	name: string;
+	description: string;
+}
+
+const StyledRoot = styled.div`
+	min-width: 500px;
+	max-width: 1200px;
+`;
+
+const EditorTargetDescription = styled.div<{ height: number }>`
+	${props => props.height ? `height: ${props.height}px` : ''};
+	margin-top: 1.3em;
+	margin-bottom: 1.3em;
+	line-height: ${props => props.theme.lineHeight};
+	font-size: 14px;
+`;
+
+const ContentRoot = styled.div`
+	background-color: ${props => props.theme.backgroundColor3};
+	padding: 1em;
+	padding-right: 0;
+`;
+
+const EditorTargetBoxes = styled.div`
+	display: flex;
+	flex-direction: row;
+	justify-content: center;
+`;
+
+const EditorTargetTitle = styled.p`
+	display: flex;
+	flex-direction: row;
+	font-weight: bold;
+	font-size: 1.7em;
+	align-items: center;
+	white-space: nowrap;
+`;
+
+const EditorTargetLogo = styled.img`
+	height: 1.3em;
+	margin-right: 0.4em;
+`;
+
+const EditorExample = styled.img`
+	width: 100%;
+	margin-top: 1em;
+`;
+
+const EditorTargetBox = styled.div`
+	display: flex;
+	flex: 1;
+	flex-direction: column;
+	font-family: ${props => props.theme.fontFamily};
+	color: ${props => props.theme.color};
+	background-color: ${props => props.theme.backgroundColor};
+	border: 1px solid ${props => props.theme.dividerColor};
+	border-radius: 8px;
+	padding: 2em 2.2em 2em 2.2em;
+	margin-right: 1em;
+	max-width: 400px;
+	opacity: 1;
+`;
+
+const SelectButton = styled(Button)`
+	padding: 10px 10px;
+    height: auto;
+    min-height: auto;
+    max-height: fit-content;
+    font-size: 1em;
+`;
+
+const ChangeEditorWarning = styled.div`
+	margin-top: 1em;
+	opacity: 0.8;
+	font-family: ${props => props.theme.fontFamily};
+	color: ${props => props.theme.color};
+	font-size: 14px;
+`;
+
+const EditorOptions: string[] = [
+	'Rich Text Editor',
+	'Markdown Editor',
+];
+
+
+const logosImageNames: Record<string, string> = {
+	'Rich Text Editor': 'default-editor_RTE_icon.png',
+	'Markdown Editor': 'default-editor_Markdown_icon.png',
+};
+
+const editorExamples: Record<string, string> = {
+	'Rich Text Editor': 'default-editor_RTE.png',
+	'Markdown Editor': 'default-editor_Markdown.png',
+};
+
+type EditorTargetInfoName = 'Rich Text Editor' | 'Markdown Editor';
+
+export default function(props: Props) {
+	const joplinCloudDescriptionRef = useRef(null);
+
+	const onButtonRowClick = useCallback(() => {
+		Setting.setValue('openDefaultEditorDialog', false);
+		props.onClose();
+	}, [props.onClose]);
+
+	const { height: descriptionHeight } = useElementSize(joplinCloudDescriptionRef);
+
+
+	const onSelectButtonClick = useCallback(async (name: EditorTargetInfoName) => {
+		const routes = {
+			'Rich Text Editor': { name: 'RTE' },
+			'Markdown Editor': { name: 'Markdown' },
+		};
+		const route = routes[name];
+
+		Setting.setValue('defaultNoteEditor', route.name);
+		Setting.setValue('openDefaultEditorDialog', false);
+		await Setting.saveAll();
```

Comment:
```suggestion
```
`Setting.saveAll` shouldn't be necessary here — `.setValue` should schedule saving settings.